Toolchain/default.nix

```diff
@@ -27,4 +27,5 @@ mkShell.override { stdenv = gcc13Stdenv; } {
     qemu
     python3
   ];
+  hardeningDisable = [ "format" ];
```

Comment:
Thanks for the PR! It looks like, before https://github.com/SerenityOS/serenity/commit/720feaa658a567e63119dfff34570d3b3d0749fb we had both "format" and "fortify" disabled. Is fortify not required to be disabled in order to build the Toolchain in a nix env?

Reply:
Yes, this seemed to make it work for me!

I don't know if it would make sense to make a CI test for it? It would be a bit expensive test to compile gcc just to see if it works, but i don't know?

I btw see that I used the "old school" `nix-shell`, while the cool kids use `nix develop`. Perhaps a bit silly to maintain both, especially since they don't produce the same exact environments: `nix develop` uses a lock file, while `nix-shell` here just uses whatever <nixpkgs> points to locally. But that's a different discussion.